src/client/response.rs

```diff
@@ -0,0 +1,42 @@
+use crate::msg::Message;
+use anyhow::{Context, Result};
+use async_stream::stream;
+use bb8_redis::{
+    bb8::{Pool, PooledConnection},
+    redis::AsyncCommands,
+    RedisConnectionManager,
+};
+use futures::Stream;
+
+pub struct Response {
+    pool: Pool<RedisConnectionManager>,
+}
+
+impl Response {
+    pub fn new(pool: Pool<RedisConnectionManager>) -> Self {
+        Self { pool }
+    }
+
+    async fn get_connection(&self) -> Result<PooledConnection<'_, RedisConnectionManager>> {
+        let conn = self
+            .pool
+            .get()
+            .await
+            .context("unable to retrieve a redis connection from the pool")?;
+
+        Ok(conn)
+    }
+
+    /// Get response from the remote/local twin
+    /// [call this after use the send function]
+    pub fn response<C: AsRef<str>>(self, ret: C) -> impl Stream<Item = Result<Message>> {
```

Comment:
This looks nice in general but there is a fundamental flow. How this will know that you have received all responses or timedout (the wait is now longer than the message expiration time)
This logic instead need to 
- If all (expected) responses has been received, there is no need to try to pop anything again.
- lf not, the deadline of all the retries can't be more than the (message timestamp + expiration)
- a Resposne object need to know about the 'timestamp, expiration, return queue' hence the respones method doesn't need to take any arguments. Again the user should know what are queues. 
- On request the message queue is set and also the timstamp of the message.